lib/spack/spack/environment/environment.py

```diff
@@ -835,10 +930,37 @@ def _construct_state_from_manifest(self):
             if "path" not in entry:
                 self.dev_specs[name]["path"] = name
 
+    def all_concretized_user_specs(self):
+        included_concretized_user_specs = []
+        for concretized_user_specs in self.included_concretized_user_specs.values():
+            included_concretized_user_specs.extend(concretized_user_specs)
+
+        return self.concretized_user_specs + included_concretized_user_specs
+
+    def all_concretized_orders(self):
+        included_concretized_order = []
+        for concretized_orders in self.included_concretized_order.values():
+            included_concretized_order.extend(concretized_orders)
+
+        return self.concretized_order + included_concretized_order
```

Comment:
```suggestion
        concretized_order = self.concretized_order[:]
        for included_concretized_order in self.included_concretized_order.values():
            concretized_order.extend(included_concretized_order)

        return concretized_order
```
This version copies each included hash once instead of twice